### DISCUSSION TITLE
Easier way to step into 2nd method on the line

### DISCUSSION THREAD

**Discussion by janko** [ruby/debug#991] [General]
When I have a line with two method calls (e.g. an inline conditional), I find it difficult to step into the 2nd method on that line.

What I'm experiencing is the following:

```sh
#   def foo
# >   bar if baz?
#     1 + 1
#   end

$ step

#   def baz?
# >   true
#   end

$ next

#   def baz?
#     true
# > end

$ next

#   def foo
#     bar if baz?
# >   1 + 1
#   end
```

I'm only able to step into `#bar` if I `step` when I'm at the `end` of `baz?`, otherwise I miss my chance. I sometimes detect this scenario and `step` when I should, but it's not easy to remember.

It would be ideal if when I go `next` after the last line of `baz?`, instead of going to `end`, I get thrown back into the `bar if baz?` line, giving me the opportunity to `step`, this time into `bar`.

**Comment by ko1** [comment]
I understand the request, but as I know most of debuggers like gdb moves to the `1+1` line.
Could you make an issue as feature request with examples?

**Comment by sbungartz** [comment]
A little late, but I just stumbled accross this: For cases where the `baz?` method would be longer or call lots of other things it might also be helpful to use `finish` to jump to the end of `baz?` and then `step` brings you to `bar`:

```
#   def foo
# >   bar if baz?
#     1 + 1
#   end

$ step

#   def baz?
# >   true
#     do_more_complicated_stuff
#   end

$ finish

#   def baz?
#     true
#     do_more_complicated_stuff
# > end

$ step

#   def bar
#     "hello world"
#   end
```

Hope this helps someone.

As an aside: In `byebug` the `finish` command got you back to the line `bar if baz?` and then `step` would also get you to `bar` but showing that line again instead of the `end` of `baz?` always felt more natural to me... 🤷
